test/functional/p2p_filter.py

```diff
@@ -93,11 +96,36 @@ def test_size_limits(self, filter_node):
 
         filter_node.send_and_ping(msg_filterclear())
 
-    def run_test(self):
-        filter_node = self.nodes[0].add_p2p_connection(FilterNode())
+    def test_msg_mempool(self):
+        self.log.info("Check that a node with bloom filters enabled services p2p mempool messages")
+        self.nodes[0].disconnect_p2ps()
+        filter_node = FilterNode()
+        filter_node.tx_received = False
 
-        self.test_size_limits(filter_node)
+        self.log.info("Create a tx relevant to the node before connecting")
+        filter_address = self.nodes[0].decodescript(filter_node.watch_script_pubkey)['addresses'][0]
+        txid = self.nodes[0].sendtoaddress(filter_address, 90)
+
+        self.log.info("Send a mempool msg after connecting and check that the tx is received")
+        self.nodes[0].add_p2p_connection(filter_node)
+        filter_node.send_and_ping(filter_node.watch_filter_init)
+        self.nodes[0].p2p.send_message(msg_mempool())
+        filter_node.wait_for_tx(txid)
+        assert filter_node.tx_received
+
+    def test_frelay_false(self, filter_node):
+        self.log.info("Check that a node with fRelay set to false does not receive invs until the filter is set")
+        filter_address = self.nodes[0].decodescript(filter_node.watch_script_pubkey)['addresses'][0]
+        filter_node.tx_received = False
+        self.nodes[0].sendtoaddress(filter_address, 90)
+        # Sync to make sure the reason filter_node doesn't receive the tx isn't due to p2p delays
+        filter_node.sync_with_ping()
+        assert not filter_node.tx_received
+
+        # Clear the mempool so that this transaction does not impact subsequent tests
+        self.nodes[0].generate(1)
 
+    def test_filter(self, filter_node):
         self.log.info('Add filtered P2P connection to the node')
```

Comment:
This log seems in the wrong place. The connection has already been created earlier.